packages/codecs-core/src/__tests__/codec-test.ts

```diff
@@ -1,75 +1,73 @@
-import { Codec, Decoder, Encoder } from '../codec';
+import { Codec, createCodec, createDecoder, createEncoder, Decoder, Encoder } from '../codec';
 
 describe('Encoder', () => {
     it('can define Encoder instances', () => {
-        const myEncoder: Encoder<string> = {
-            description: 'myEncoder',
-            encode: (value: string) => {
-                const bytes = new Uint8Array(32).fill(0);
+        const myEncoder: Encoder<string> = createEncoder({
+            fixedSize: 32,
+            write: (value: string, bytes, offset) => {
                 const charCodes = [...value.slice(0, 32)].map(char => Math.min(char.charCodeAt(0), 255));
-                bytes.set(new Uint8Array(charCodes));
-                return bytes;
+                bytes.set(charCodes, offset);
+                return offset + 32;
             },
```

Comment:
Is there any reason to make this context-y with `this`?
```suggestion
            write(value: string, bytes, offset) {
                const length = this.fixedSize;
                const charCodes = [...value.slice(0, length)].map(char => Math.min(char.charCodeAt(0), 255));
                bytes.set(charCodes, offset);
                return offset + length;
            },
```